examples/msal-go/token_credential.go

```diff
@@ -4,83 +4,72 @@ import (
 	"context"
```

Comment:
Do we have tests for this example?

Reply:
This example is built and used as part of our e2e tests (ref: https://github.com/Azure/azure-workload-identity/blob/main/test/e2e/token_exchange.go)